api/server/objectstore.go

```diff
@@ -0,0 +1,140 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/libopenstorage/openstorage/objectstore"
+)
+
+// swagger:operation GET /cluster/objectstore objectstore objectStoreInspect
+//
+// Lists Objectstore
+//
+// This will list current object stores
+//
+// ---
+// produces:
+// - application/json
+// responses:
+//   '200':
+//     description: success
+//     schema:
+//      $ref: '#/definitions/ObjectstoreInfo'
+func (c *clusterApi) objectStoreInspect(w http.ResponseWriter, r *http.Request) {
+	method := "objectStoreInspect"
+
+	objInfo, err := c.ObjectStoreManager.ObjectStoreInspect()
+	if err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	json.NewEncoder(w).Encode(objInfo)
+}
+
+// swagger:operation POST /cluster/objectstore objectstore objectStoreCreate
+//
+// Create an Object store
+//
+// This creates the volumes required to run the object store
+//
+// ---
+// produces:
+// - application/json
+// parameters:
+// - name: name
+//   in: query
+//   description: volume on which object store to run
+//   required: true
+//   type: string
+// responses:
+//   '200':
+//     description: success
+func (c *clusterApi) objectStoreCreate(w http.ResponseWriter, r *http.Request) {
+	method := "objectStoreCreate"
+	params := r.URL.Query()
+	volumeName := params[objectstore.VolumeName]
+
+	if len(volumeName) == 0 || volumeName[0] == "" {
+		c.sendError(c.name, method, w, "Missing secret ID", http.StatusBadRequest)
+		return
+	}
+
+	err := c.ObjectStoreManager.ObjectStoreCreate(volumeName[0])
```

Comment:
I would assume that when ObjectStorageCreate is called with a volume ID, that it returns the object storage information on how to access it. Why does it not return that information?

Reply:
There is Inspect call for retrieving information about objectstore. Objectstorecreate creates object store, you have to call Objectstoreupdate(true) to enable it.

Current PX Create Object Store API does not send back info about objectstore in create call, if I voluntarily add it here I have to add Inspect() call inside Create() call while integration

Reply:
It should return some type of information. We should do the right thing and return the requested data information to the caller. Please work with others to make sure we can change the interfaces also (yes, we should change the interfaces) to make sure that they return the appropriate information.

Reply:
Creating an objectstore just created a binding between the objectstore and the volume to be used. The credentials are created when the objectstore is first enabled, so the only thing you can return here is the objectstore UUID, which isn't very helpful right now since there can only be one objectstore anyways.